src/umockcallrecorder.c

```diff
@@ -366,68 +481,75 @@ const char* umockcallrecorder_get_actual_calls(UMOCKCALLRECORDER_HANDLE umock_ca
         size_t i;
         char* new_actual_calls_string;
 
-        if (umock_call_recorder->actual_call_count == 0)
+        /* Codes_SRS_UMOCKCALLRECORDER_01_073: [ If a lock was created for the call recorder, umockcallrecorder_get_actual_calls shall acquire the lock in shared mode. ]*/
+        internal_lock_acquire_shared_if_needed(umock_call_recorder);
         {
-            new_actual_calls_string = (char*)umockalloc_realloc(umock_call_recorder->actual_calls_string, 1);
-            if (new_actual_calls_string == NULL)
+            if (umock_call_recorder->actual_call_count == 0)
             {
-                /* Codes_SRS_UMOCKCALLRECORDER_01_026: [ If allocating memory for the resulting string fails, umockcallrecorder_get_actual_calls shall fail and return NULL. ]*/
-                UMOCK_LOG("umockcallrecorder: Cannot allocate memory for actual calls.");
-                result = NULL;
+                new_actual_calls_string = umockalloc_realloc(umock_call_recorder->actual_calls_string, 1);
```

Comment:
>mockalloc_realloc(umock_call_recorder->actual_calls_string [](start = 43, length = 58)

Also seems like exclusive would be needed here

Reply:
Fixed.

---
In reply to: [530654384](https://github.com/Azure/umock-c/pull/177#discussion_r530654384) [](ancestors = 530654384)